app/src/main/java/com/github/orkest/View/EditProfileActivity.kt

```diff
@@ -0,0 +1,221 @@
+ * - current profile picture
+ * - "edit picture" button to modify picture
+ */
+@Composable
+fun topBar() {
+    Column(
+        modifier = Modifier
+            .verticalScroll(rememberScrollState())
+            .padding(8.dp)
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(8.dp),
+            // this spaces out the "cancel" and "save" buttons
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            // "cancel" clickable text (button)
+            Text(
+                text = "Cancel",
+                modifier = Modifier.clickable { /* TODO */ }
+            )
+
+            // "save" clickable text (button)
+            Text(
+                text = "Save",
+                modifier = Modifier.clickable { /* TODO */ }
+            )
+        }
+        // profile pic and edit button
+        EditProfileImage()
+    }
+}
+
+/**
+ * The larger part of the screen containing the fields to modify textual information
+ */
+@Composable
+fun mainBody() {
+    EditNameSection(name = "Name", default = "default name")
+    EditNameSection(name = "Username", default = "default username")
+    EditBio()
+}
+
+/**
+ * Display current profile picture and an "edit picture" button that is used to choose a new one
+ */
+@Composable
+fun EditProfileImage() {
+    val imageUri = rememberSaveable { mutableStateOf("") }
+    val painter = rememberImagePainter(
+        imageUri.value.ifEmpty { R.drawable.blank_profile_pic }
+    )
+
+    // will be used to access the images library on your phone and set a new picture
+    val launcher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.GetContent())
+    {
+        uri: Uri? -> uri?.let { imageUri.value = it.toString() }
+    }
+
+    Column(
+        modifier = Modifier
+            .padding(8.dp)
+            .fillMaxWidth(),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        // displays your profile picture in a circle
+        Card(
+            shape = CircleShape,
+            modifier = Modifier
+                .padding(8.dp)
+                .size(100.dp))
+        {
+            Image(
+                painter = painter,
+                contentDescription = null,
+                modifier = Modifier.wrapContentSize(),
+                contentScale = ContentScale.Crop)
+        }
+        // clickable Text offering the possibility to change profile pic
+        Text(
+            text = "edit picture",
+            modifier = Modifier.clickable { launcher.launch("image/*") })
+    }
+}
+
+/**
+ * fields to modify small text data such as name and username
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EditNameSection(name: String, default: String) {
+    var modifyName by rememberSaveable { mutableStateOf(default) }
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 4.dp, end = 4.dp),
```

Comment:
It's just a design opinion but I would use a large padding so that the fields aren't too close to the borders of the screen